extensions/data-workspace/src/test/workspaceService.test.ts

```diff
@@ -225,6 +232,44 @@ suite('WorkspaceService Tests', function (): void {
 		onWorkspaceProjectsChangedDisposable.dispose();
 	});
 
+	test('test addProjectsToWorkspace when no workspace open', async () => {
+		stubWorkspaceFile(undefined);
+		const onWorkspaceProjectsChangedStub = sinon.stub();
+		const onWorkspaceProjectsChangedDisposable = service.onDidWorkspaceProjectsChange(() => {
+			onWorkspaceProjectsChangedStub();
+		});
+		sinon.stub(workspace, 'createWorkspace').resolves(undefined);
+
+		await service.addProjectsToWorkspace([
+			vscode.Uri.file('/test/folder/proj1.sqlproj')
+		]);
+
+		should.strictEqual(onWorkspaceProjectsChangedStub.calledOnce, false, 'the onDidWorkspaceProjectsChange event should not have been fired');
```

Comment:
calledOnce is false if it is called multiple times, here you should use notCalled, also you should verify createWorkspace stub is called once.

Reply:
did you forget to push you local changes? this one should be fixed, right?

Reply:
oh no I did forget to push my local changes. Here's another PR to address the last comments https://github.com/microsoft/azuredatastudio/pull/13004